src/src/com/microsoft/aad/adal/IdToken.java

```diff
@@ -23,28 +23,145 @@
 
 package com.microsoft.aad.adal;
 
+import java.io.UnsupportedEncodingException;
+import java.util.HashMap;
+import java.util.Iterator;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import android.util.Base64;
+
 class IdToken {
```

Comment:
This contains all the claims we get from idtoken. UserInfo is a subset, only contains those info represents a single user, i.e. tenant id won't be in the userInfo object.